analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StreamReadStatement.cs

```diff
@@ -14,19 +14,18 @@ public StreamReadStatement(string fileName)
             using (var stream = File.Open(fileName, FileMode.Open))
             {
                 var result = new byte[stream.Length];
-                stream.Read(result, 0, (int)stream.Length); // Noncompliant
+                stream.Read(result, 0, (int)stream.Length);            // Noncompliant
 //              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                 var l = stream.Read(result, 0, (int)stream.Length);
-                stream.ReadAsync(result, 0, (int)stream.Length); // Noncompliant {{Check the return value of the 'ReadAsync' call to see how many bytes were read.}}
+                stream.ReadAsync(result, 0, (int)stream.Length);       // Noncompliant {{Check the return value of the 'ReadAsync' call to see how many bytes were read.}}
                 await stream.ReadAsync(result, 0, (int)stream.Length); // Error [CS4033] - ctor can't be async // Noncompliant
                 stream.Write(result, 0, (int)stream.Length);
             }
-            streamField.Read(new byte[0], 0, 0); // Noncompliant
-            streamField?.Read(new byte[0], 0, 0); // Noncompliant
-            Read();
+            streamField.Read(new byte[0], 0, 0);                       // Noncompliant
+            this.streamField.Read(new byte[0], 0, 0);                  // Noncompliant
+            streamField?.Read(new byte[0], 0, 0);                      // Noncompliant
+            this.streamField?.Read(new byte[0], 0, 0);                 // Noncompliant
         }
-
-        private void Read() { }
```

Comment:
Why remove that?

Reply:
I can't remember exactly, but I thought `Read` methods were not checked correctly. So I added a test for it in 35dd2f5f924343523888aa0f57701ff59dbfcb4f. Later I discovered code covered unrelated `Read` methods but I don't think tests are in place. I will restore the test. Thanks for catching this.